python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -24,6 +24,7 @@ def set_index_post(df, index_name, drop, column_dtype):
 
 def _set_partitions_pre(s, divisions):
     partitions = divisions.searchsorted(s, side="right") - 1
+    partitions[partitions < 0] = 0
     partitions[
         divisions.tail(1).searchsorted(s, side="right").astype("bool")
```

Comment:
Could we get away with skipping this computation and doing something like

```suggestion
        partitions >= len(divisions) - 1
```

That would be in line with the additional check I made, and seems less computationally expensive.

Reply:
Yeah - My memory is a bit rusty on the logic here, but it does seem like the existing logic is "excessive" :)

It looks like we are taking the last division and checking if any value in `s` is equal to or greater than that final division.  For the elements of `s` that are beyond the last division, we explicitly set their partition. If tests pass, then I don't see any reason why we cannot just focus the logic on the partitions themselves (as you suggest).

Reply:
Looks like this change is non-breaking - pushing it in! Thanks 🙂